Prevent links being converted to relative links

wr2f25 opened the pull request:
This will need similar changes in 3.5 core in order to work with the JSON richtext.

Fixes #6.

wr2f25 committed:
Prevent links being converted to relative links

Fixes #6.

drn05r reviewed (commented):
Reviewing https://www.tiny.cloud/docs/tinymce/latest/url-handling/ `relative_url` will fix issues where a user adds something with a relative link which then could be wrong in a different context.

It looks like the links may still be absolute paths rather than full URLs, which mean there could be issues if richtext content is embedded but #6 is only concerned with richtext within the context of the EPrints repository.  If it needed to be embedded then it would be on the user to specify full URLs in any links they add. 

I assume if a user added a full URL where the hostname match the current host TinyMCE would not convert this into a absolute path rather than a full URL?

wr2f25 committed:
Store the full domain path for local links

Previously it would create an absolute path from the domain root however
this would mean if it ended up embedded on another site for some reason
it would break. As such this changes it so that it will also add the
domain to the URL.
